vehicle_gateway_python_helpers/vehicle_gateway_python_helpers/helpers.py

```diff
@@ -0,0 +1,157 @@
+# Copyright 2023 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from distutils.dir_util import copy_tree
+
+from ament_index_python.packages import get_package_share_directory
+from launch import Substitution, SomeSubstitutionsType, LaunchContext
+
+import xml.etree.ElementTree as ET
+import os
+from typing import List
+
+
+def get_px4_dir():
+    return get_package_share_directory('px4_sim')
+
+
+def seed_rootfs(rootfs):
+    px4_dir = get_px4_dir()
+    print(f'seeding rootfs at {rootfs} from {px4_dir}')
+    copy_tree(px4_dir, rootfs)
+
+
+class WorldPoseFromSdfFrame(Substitution):
+    """Substitution that retrieves a frame from SDF."""
+
+    def __init__(self,
+                 frame_name: SomeSubstitutionsType,
+                 world_name: SomeSubstitutionsType,
+                 model_pose: SomeSubstitutionsType,
+                 coor_name: str) -> None:
+        super().__init__()
+
+        from launch.utilities import normalize_to_list_of_substitutions  # import here to avoid loop
+        self.__frame_name = normalize_to_list_of_substitutions(frame_name)
+        self.__world_name = normalize_to_list_of_substitutions(world_name)
+        self.__model_pose = normalize_to_list_of_substitutions(model_pose)
+        self.__x = "0.0"
+        self.__y = "0.0"
+        self.__z = "0.3"
+        self.__roll = "0.0"
+        self.__pitch = "0.0"
+        self.__yaw = "0.0"
+        self.__coord_name = coor_name
+
+    @property
+    def model_pose(self) -> List[Substitution]:
+        """Getter for model pose."""
+        return self.__model_pose
+
+    @property
+    def frame_name(self) -> List[Substitution]:
+        """Getter for frame name."""
+        return self.__frame_name
+
+    @property
+    def world_name(self) -> List[Substitution]:
+        """Getter for world name."""
+        return self.__world_name
+
+    def parseCoords(self, strCoords: str, key: str, strSplit: str):
+        x, y, z, roll, pitch, yaw = strCoords.split(strSplit)
+        if (self.__coord_name == "x"):
+            return str(x)
+        if (self.__coord_name == "y"):
+            return str(y)
+        if (self.__coord_name == "z"):
+            return str(z)
+        if (self.__coord_name == "roll"):
+            return str(roll)
+        if (self.__coord_name == "pitch"):
+            return str(pitch)
+        if (self.__coord_name == "yaw"):
+            return str(yaw)
+        return "0.0"
```

Comment:
I think you meant to use "key" instead of `self.__coord_name`. Otherwise, `key` doesn't seem to be used. 

I would also think it's good to use the Python literal type, since you know all the expected types here: 
```
key: Literal["x", "y", "z", "roll", "pitch", "yaw"]
```
https://docs.python.org/3/library/typing.html#typing.Literal